### PR TITLE
test: Fix the instability issue in ASM E2E tests, as well as the failure issue in asm-cost-center.e2e

### PR DESCRIPTION
**Pull request by luoyifeng11**
The primary objective is to resolve the failures in ASM E2E tests. For a comprehensive analysis of the reasons and the goals of the adjustments, please refer to the ticket: [CXSPA-3009](https://jira.tools.sap/browse/CXSPA-3009)

### CODE REVIEW

**Review by i53577 — DISMISSED** [review]
LGTM

**Review by niehuayang — DISMISSED** [review]

**Review by niehuayang — DISMISSED** [review]

File: projects/storefrontapp-e2e-cypress/package.json [hunk 1]
```diff
@@ -17,10 +17,10 @@
...
-    "cy:run:ci:b2b": "cypress run --config-file cypress.config.ci.ts --env BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers --record --key $CYPRESS_KEY --tag \"b2b,all\" --parallel --ci-build-id $BUILD_NUMBER --group B2B --spec \"cypress/e2e/b2b/regression/**/*e2e.cy.ts\" --reporter junit --reporter-options mochaFile=results/e2e-test-b2b-[hash].xml",
-    "cy:run:ci:core:b2b": "cypress run --config-file cypress.config.ci.ts --env BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers --record --key $CYPRESS_KEY --tag \"b2b,core\" --parallel --ci-build-id $BUILD_NUMBER --group B2B --spec \"cypress/e2e/b2b/regression/**/*.core-e2e.cy.ts\"",
+    "cy:run:ci": "cypress run --config-file cypress.config.ci.ts --record --key $CYPRESS_KEY --tag \"b2c,all\" --parallel --ci-build-id $BUILD_NUMBER --group B2C --spec \"cypress/e2e/regression/asm/**/*e2e.cy.ts\" --reporter junit --reporter-options mochaFile=results/e2e-test-b2c-[hash].xml",
```
Comment (niehuayang) [inline projects/storefrontapp-e2e-cypress/package.json]: If I understand correctly,  we don't should change the script.

File: projects/storefrontapp-e2e-cypress/package.json [hunk 2]
```diff
@@ -17,10 +17,10 @@
...
+    "cy:run:ci:core": "cypress run --config-file cypress.config.ci.ts --record --key $CYPRESS_KEY --tag \"b2c,core\" --parallel --ci-build-id $BUILD_NUMBER --group B2C --spec \"cypress/e2e/regression/asm/**/*.core-e2e.cy.ts\"",
```
Comment (niehuayang) [inline projects/storefrontapp-e2e-cypress/package.json]: If I understand correctly,  we don't should change the script.

File: projects/storefrontapp-e2e-cypress/package.json [hunk 3]
```diff
@@ -17,10 +17,10 @@
...
+    "cy:run:ci:b2b": "cypress run --config-file cypress.config.ci.ts --env BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers --record --key $CYPRESS_KEY --tag \"b2b,all\" --parallel --ci-build-id $BUILD_NUMBER --group B2B --spec \"cypress/e2e/b2b/regression/asm/**/*e2e.cy.ts\" --reporter junit --reporter-options mochaFile=results/e2e-test-b2b-[hash].xml",
+    "cy:run:ci:core:b2b": "cypress run --config-file cypress.config.ci.ts --env BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers --record --key $CYPRESS_KEY --tag \"b2b,core\" --parallel --ci-build-id $BUILD_NUMBER --group B2B --spec \"cypress/e2e/b2b/regression/asm/**/*.core-e2e.cy.ts\"",
```
Comment (niehuayang) [inline projects/storefrontapp-e2e-cypress/package.json]: If I understand correctly,  we don't should change the script.

File: projects/storefrontapp-e2e-cypress/cypress/e2e/b2b/regression/checkout/asm-cost-center.e2e.cy.ts [hunk 4]
```diff
@@ -48,47 +48,21 @@ context('B2B - ASM Account Checkout', () => {
...
-      cy.get('select').select(invalid_cost_center);
-    });
-
-    cy.wait('@costCenterReq').its('response.statusCode').should('eq', 400);
-    alerts.getErrorAlert().contains('Invalid cost center.');
-    alerts.getErrorAlert().then((alert) => {
-      cy.wrap(alert).within(() => {
-        cy.get('button').click();
+      cy.get('select').then((select) => {
+        cy.get('select').select(valid_cost_center);
```
Comment (niehuayang) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/b2b/regression/checkout/asm-cost-center.e2e.cy.ts]: Please explain why we need to change this

Comment (luoyifeng11) [reply projects/storefrontapp-e2e-cypress/cypress/e2e/b2b/regression/checkout/asm-cost-center.e2e.cy.ts]: The main reason for the failure of running asm-cost-center.e2e.ts is that in the B2B scenario, during the payment information filling step, it is necessary to select a cost center. 
However, currently, the dropdown menu for cost centers only contains "valid cost center," so the previous code needs to select an "invalid cost center," which results in an error. 
Therefore, I have modified the logic for this condition.

File: projects/storefrontapp-e2e-cypress/package.json (hunk 3 above)
Comment (luoyifeng11) [reply projects/storefrontapp-e2e-cypress/package.json]: I forgot to revert this change back mainly for the purpose of testing only the asm-related tests. 
Running all the tests takes too long, so I will restore it back once the testing is completed.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/asm.ts [hunk 5]
```diff
@@ -136,22 +136,24 @@ export function listenForCustomerCreateRequest(): string {
...
-    cy.get('cx-customer-selection').should('not.exist');
-    cy.get('cx-csagent-login-form form').within(() => {
-      cy.get('[formcontrolname="userId"]').should('not.be.disabled').type(user);
-      cy.get('[formcontrolname="password"]')
-        .should('not.be.disabled')
-        .type(pwd);
-      cy.get('button[type="submit"]').click();
-    });
-  });
+  cy.get('cx-storefront cx-csagent-login-form').then(($element) => {
```
Comment (luoyifeng11) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/asm.ts:139]: If the agent is already logged in, there is no need to log in again.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/saved-cart.ts [hunk 6]
```diff
@@ -247,12 +247,12 @@ export function saveActiveCart(verify = true) {
...
 
       cy.get('cx-saved-cart-form-dialog').within(() => {
-        cy.get('[formcontrolname="name"]').type(
-          sampleData.savedActiveCartForm[0].name
-        );
-        cy.get('[formcontrolname="description"]').type(
-          sampleData.savedActiveCartForm[0].description
-        );
+        cy.get('[formcontrolname="name"]')
+          .clear()
```
Comment (luoyifeng11) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/saved-cart.ts:251]: During the pipeline testing process, when opening the "Save Cart" dialog, if there is already text present in the input field, it results in the typed content being appended after the existing text, leading to testing errors.

**Review by niehuayang — DISMISSED** [review]
LGTM

**Review by niehuayang — APPROVED** [review]

**Review by kimhw0630 — APPROVED** [review]

**Pull request merged**